src/Utilities/ErrorHandling/FormatStacktrace.cpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Utilities/ErrorHandling/FormatStacktrace.hpp"
+
+#include <array>
+#include <boost/algorithm/string.hpp>  // For boost::to_lower
+#include <boost/core/demangle.hpp>
+#include <cstddef>
+#include <cstdlib>     // For std::getenv
+#include <execinfo.h>  // For backtrace_symbols
+// link.h is not available on all platforms
+#if __has_include(<link.h>)
+#include <link.h>
+#endif
+#include <memory>
+#include <string>
+
+namespace stacktrace_detail {
+
+namespace {
+bool env_var_to_bool(std::string env_var) {
+  boost::to_lower(env_var);
+  return not(env_var == "false" or env_var == "no");
+}
+}  // namespace
+
+std::string abbreviated_symbol_name(const std::string& symbol_name) {
+  constexpr size_t abbrev_length_begin = 300;
+  constexpr size_t abbrev_length_end = 100;
+  // Length of " [...] "
+  constexpr size_t abbrev_length_separator = 7;
+  if (symbol_name.size() <=
+      abbrev_length_begin + abbrev_length_end + abbrev_length_separator) {
+    return symbol_name;
+  }
+  // Allow an environment variable to toggle the abbreviation
+  const char* enable_abbrev_env =
+      std::getenv("SPECTRE_ABBREV_BACKTRACE_SYMBOLS");
+  if (enable_abbrev_env != nullptr and
```

Comment:
I believe most Linux tools just check if the variable is set, not what it is set to. The value only matters if it's controlling a non-boolean thing, though sometimes I think `0` and `1` are used for boolean, but then there's the difference bash and C using different int->bool truth values (C is wrong, unfortunately :P ).

@wthrowe thoughts on this?

Reply:
I looked around a bit.  All the old core things (like bash, df) only check whether the variable is set.  Perl seems to like to check for the value "1".  As we've seen before, hdf5 uses the exact string "FALSE" (default is on, unlike the others).  Python uses "set and nonempty" (the unset vs. set-and-empty distinction can be a bit weird and annoying).

SpEC does at least two different things, of course.  For backtraces it uses set-and-nonempty like python.  For CPU binding it uses unset = "0" = "no" = "false" and "1" = "yes" = "true" (all case insensitive), with other values causing an error.  (Same function as for parsing booleans from input files.)

So no agreement.  In the absence of any other constraints, I like the python method.  (I think I wrote the SpEC one that does that.)  No problem remembering which values are special, and no need to remember how to use `env` to unset variables if I want it temporarily off.

The added difficulty here is that the option defaults to on.  As much as I generally dislike such things, I'd recommend a negative option here, since I find the empty string being true to be unexpected.

Reply:
I have no strong opinion here, so happy to do what you recommend @wthrowe. What I find strange about the "set and nonempty" method though is that I don't know what to actually set the variable to? Setting it to 1 or 0 or true or false all has the same effect, which is weird for a boolean value. Therefore I kind of like the behavior you described for SpEC's CPU binding. Also means we don't have to make it a negative option. What do you think?

Reply:
We can't error on an invalid value inside the backtrace printer.

Reply:
Well ok I did the "set and nonempty". I find the bool(0) == bool(1) thing strange, but not important enough to worry about it any longer :)

Reply:
I'm not opposed to any of the others (like checking for "1") if you prefer them, nonempty is just my favorite.  Just that, as a practical manner, our error handler isn't allowed to fail, so we have to accept any input we get.